### PR TITLE
make max inflight warning global to all pipelines

### PR DESCRIPTION
**Pull request by jsvd**
```yml            
- pipeline.id: test_01
  config.string: "input { heartbeat {} } filter { sleep { time => 1 } } output { stdout { codec => dots } }"
- pipeline.id: test_02
  config.string: "input { heartbeat {} } filter { sleep { time => 1 } } output { stdout { codec => dots } }"
- pipeline.id: test_03
  config.string: "input { heartbeat {} } filter { sleep { time => 1 } } output { stdout { codec => dots } }"
```

```
~/elastic/logstash improve_max_inflight_warning ❯ bin/logstash -r -b 125 --debug | grep resource                 
[2024-10-24T11:44:44,941][DEBUG][logstash.pipelineresourceusagevalidator] For a baseline of 2KB events, the maximum heap memory consumed across 3 pipelines may reach up to 0.72% of the entire heap (more if the events are bigger).
```

```
~/elastic/logstash improve_max_inflight_warning ❯ bin/logstash -r -b 12500 --debug | grep resource
[2024-10-24T11:45:13,008][WARN ][logstash.pipelineresourceusagevalidator] For a baseline of 2KB events, the maximum heap memory consumed across 3 pipelines may reach up to 71.53% of the entire heap (more if the events are bigger). The recommended percentage is less than 10%. Consider reducing the number of pipelines, or the batch size and worker count per pipeline.
```

closes https://github.com/elastic/logstash/issues/8622

### CODE REVIEW

File: logstash-core/lib/logstash/pipeline_resource_usage_validator.rb [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    include LogStash::Util::Loggable
+
+    WARN_HEAP_THRESHOLD = 10 # 10%
+
+    def initialize(max_heap_size)
+      @max_heap_size = max_heap_size
+    end
+
+    # TODO
+    # @param pipelines_registry
```
Comment (jsvd) [inline logstash-core/lib/logstash/pipeline_resource_usage_validator.rb:29]:
```suggestion
```

File: logstash-core/lib/logstash/pipeline_resource_usage_validator.rb [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+      end
+    end
+
+    def compute_percentage(pipelines_registry)
+      max_event_count = sum_event_count(pipelines_registry)
+      estimated_heap_usage = max_event_count * 2.0 * 1024 # assume 2KB per event
+      percentage_of_heap = ((estimated_heap_usage / @max_heap_size) * 100).round(2)
+    end
+
+    def sum_event_count(pipelines_registry)
```
Comment (andsel) [inline logstash-core/lib/logstash/pipeline_resource_usage_validator.rb:46]: This method iterates on the state exposed by `pipeline_registry`, what do you think if this should be part of the `PipelineRegistry` class?

**Review by andsel — APPROVED** [review]
LGTM left just a doubt if iterating over the list of pipelines should be something internal to the `PipelineRegistry` itself.

**Comment by jsvd** [comment]
@logstashmachine backport 8.x

**Pull request merged**